From: Carlos Garnacho <carlosg@gnome.org>
Date: Fri, 5 Jun 2020 16:55:02 +0200
Subject: clutter: Drop clutter_input_device_get_[pointer_]stage()

Input devices are not related to the stage in any way. Drop all the
users that relied on it being so.

https://gitlab.gnome.org/GNOME/mutter/-/merge_requests/1486
---
 clutter/clutter/clutter-input-device-private.h |  2 --
 clutter/clutter/clutter-input-device.c         | 33 --------------------------
 clutter/clutter/clutter-input-device.h         |  2 --
 3 files changed, 37 deletions(-)

diff --git a/clutter/clutter/clutter-input-device-private.h b/clutter/clutter/clutter-input-device-private.h
index 11e0c7f..138e12b 100644
--- a/clutter/clutter/clutter-input-device-private.h
+++ b/clutter/clutter/clutter-input-device-private.h
@@ -178,8 +178,6 @@ CLUTTER_EXPORT
 void clutter_input_device_update_from_tool (ClutterInputDevice     *device,
                                             ClutterInputDeviceTool *tool);
 CLUTTER_EXPORT
-ClutterStage * _clutter_input_device_get_stage (ClutterInputDevice *device);
-CLUTTER_EXPORT
 void _clutter_input_device_set_stage (ClutterInputDevice *device,
                                       ClutterStage       *stage);
 CLUTTER_EXPORT
diff --git a/clutter/clutter/clutter-input-device.c b/clutter/clutter/clutter-input-device.c
index 6f4f9d4..9244829 100644
--- a/clutter/clutter/clutter-input-device.c
+++ b/clutter/clutter/clutter-input-device.c
@@ -655,20 +655,6 @@ _clutter_input_device_set_stage (ClutterInputDevice *device,
    */
 }
 
-/*< private >
- * clutter_input_device_get_stage:
- * @device: a #ClutterInputDevice
- *
- * Retrieves the stage currently associated with @device.
- *
- * Return value: The stage currently associated with @device.
- */
-ClutterStage *
-_clutter_input_device_get_stage (ClutterInputDevice *device)
-{
-  return device->stage;
-}
-
 static void
 _clutter_input_device_free_touch_info (gpointer data)
 {
@@ -1076,25 +1062,6 @@ clutter_input_device_get_actor (ClutterInputDevice   *device,
   return info->actor;
 }
 
-/**
- * clutter_input_device_get_pointer_stage:
- * @device: a #ClutterInputDevice of type %CLUTTER_POINTER_DEVICE
- *
- * Retrieves the #ClutterStage underneath the pointer of @device
- *
- * Return value: (transfer none): a pointer to the #ClutterStage or %NULL
- *
- * Since: 1.2
- */
-ClutterStage *
-clutter_input_device_get_pointer_stage (ClutterInputDevice *device)
-{
-  g_return_val_if_fail (CLUTTER_IS_INPUT_DEVICE (device), NULL);
-  g_return_val_if_fail (device->device_type == CLUTTER_POINTER_DEVICE, NULL);
-
-  return device->stage;
-}
-
 /**
  * clutter_input_device_get_device_name:
  * @device: a #ClutterInputDevice
diff --git a/clutter/clutter/clutter-input-device.h b/clutter/clutter/clutter-input-device.h
index dd4512a..536cf8c 100644
--- a/clutter/clutter/clutter-input-device.h
+++ b/clutter/clutter/clutter-input-device.h
@@ -95,8 +95,6 @@ CLUTTER_EXPORT
 ClutterActor *          clutter_input_device_get_actor (ClutterInputDevice   *device,
                                                         ClutterEventSequence *sequence);
 CLUTTER_EXPORT
-ClutterStage *          clutter_input_device_get_pointer_stage  (ClutterInputDevice  *device);
-CLUTTER_EXPORT
 const gchar *           clutter_input_device_get_device_name    (ClutterInputDevice  *device);
 CLUTTER_EXPORT
 ClutterInputMode        clutter_input_device_get_device_mode    (ClutterInputDevice  *device);
